core/tests/test_utils.py

```diff
@@ -200,6 +202,216 @@ def get_storage_model_classes():
                     yield clazz
 
 
+class PopenStub(python_utils.OBJECT):
+    """Stubs the API of psutil.Popen() to make unit tests less expensive.
```

Comment:
This is a very big stub, which worries me a bit.

Can you explain (in some detail) what exactly makes the unit tests expensive if we don't stub it? Also, what is the time/resource savings -- can that be quantified in any way?

Reply:
Starting a new process for every unit test is intrinsically more expensive than checking an object's attributes, and for some developers it isn't even possible for them to kill them afterwards (due to permission errors).

develop's `scripts.common_test` runs 78 tests in 50.7 seconds
This PR's `scripts.common_test` runs 97 tests in 32.3 seconds

So using this mock gives us a boost of ~4.62x per-test.